lib/collection/proxy-config.js

```diff
@@ -102,176 +88,18 @@ _.assign(ProxyConfig.prototype, /** @lends ProxyConfig.prototype */ {
             return;
         }
         _.has(options, 'server') && this.server.update(options.server);
-        _.isString(options.match) && (this.match = options.match);
+        _.isString(options.match) && (this.match = new UrlMatchPattern(options.match));
```

Comment:
I thought of the same too, but I did not go down this path, because a `UrlMatchMattern` is a generic class. It takes a pattern and tests input Urls on it. I am not sure, if adding restrictions on what is allowed in the pattern or allowing a feature for it to be customisable is out of the scope of `UrlMatchPattern`.